tests/NuGet.Services.BasicSearchTests/Fixtures/StartupTestsFixture.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using NuGet.Services.BasicSearch;
+using NuGet.Services.BasicSearchTests.TestSupport;
+
+namespace NuGet.Services.BasicSearchTests.Fixtures
+{
+    public class StartupTestsFixture : IDisposable
+    {
+        private readonly Lazy<StartedWebApp<Startup>> _lazyApp;
+
+        public StartedWebApp<Startup> App => _lazyApp.Value;
+
+        public StartupTestsFixture()
+        {
+            _lazyApp = new Lazy<StartedWebApp<Startup>>(() => InitializeAsync().Result);
+        }
+
+        public void Dispose()
+        {
+            if (_lazyApp.IsValueCreated)
+            {
+                _lazyApp.Value.Dispose();
+            }
+        }
+
+        private async Task<StartedWebApp<Startup>> InitializeAsync()
+        {
+            // list packages that each test needs
+            var packageVersions = new Dictionary<string, IEnumerable<PackageVersionIdentifier>>
+            {
+                {
+                    nameof(StartupTests.AutocompleteParameterIsObserved),
+                    new[]
+                    {
+                        new PackageVersionIdentifier("knockoutjs", "2.0.0-rc"),
+                        new PackageVersionIdentifier("Modernizr", "2.8.3"),
+                        new PackageVersionIdentifier("reactiveui", "5.5.1"),
+                        new PackageVersionIdentifier("reactiveui", "5.99.6-beta")
+                    }
+                }
+            };
+
+            // construct the test settings
+            var settings = new BasicSearchTestSettings
+            {
+                PackageSource = new Uri("https://nuget.org/api/v2/"),
+                PackageVersions = packageVersions.SelectMany(p => p.Value).Distinct().ToArray(),
+                PackageDirectory = GetDirectoryPath("Packages"),
+                LuceneDirectory = GetDirectoryPath("Lucene"),
+                DataDirectory = GetDirectoryPath("Data"),
+                RegistrationBaseAddress = "http://api.nuget.org/v3/registration0/"
+            };
+
+            // initialize the Lucene and Data directories
+            await new LuceneInitializer(settings).InitializeAsync();
+            await new DataInitializer(settings).InitializeAsync();
+
+            // establish configuration for the web service
+            ConfigurationManager.AppSettings.Set("Local.Lucene.Directory", settings.LuceneDirectory);
```

Comment:
My vote is on using the approach we have in the gallery functional tests: use a `ConfigurationProvider` that does a fallback:

* Environment variable with that name set? Use environment variable.
* No environment variable but a setting exists? Use setting.
* None set? `null`

This makes it easier to change the test setting during a run. Optionally a test setup/teardown can set/unset a variable as well.